sqlglot/parser.py

```diff
@@ -1329,6 +1329,27 @@ class Parser(metaclass=_Parser):
     # The style options for the DESCRIBE statement
     DESCRIBE_STYLES = {"ANALYZE", "EXTENDED", "FORMATTED", "HISTORY"}
 
+    # The style options for the ANALYZE statement
+    ANALYZE_STYLES = {
+        "VERBOSE": lambda self: "VERBOSE",
+        "FULL": lambda self: "FULL",
+        "SAMPLE": lambda self: "SAMPLE",
+        "SKIP_LOCKED": lambda self: "SKIP_LOCKED",
+        "LOCAL": lambda self: "LOCAL",
+        "NO_WRITE_TO_BINLOG": lambda self: "NO_WRITE_TO_BINLOG",
+        "BUFFER_USAGE_LIMIT": lambda self: f"BUFFER_USAGE_LIMIT {self._parse_number()}",
+    }
```

Comment:
This looks too repetitive, let's get rid of it. For the last one you can use a branch to `_parse_number` if you match `BUFFER_USAGE_LIMIT`.